**Week 1 683380579-6 นางสาวธนิษฐา อภิรัตน์มนตรี**

Lab 1.1: Environment Setup & First Script

In [2]:
print("Hello, World!")

Hello, World!


Lab 1.2: Working with Variables and Basic Data Types
- Objective: Students will define variables of different data types, perform simple operations, and display results.


2. Declare and print variables:


In [4]:
Age = 21
GPA = 4.00
Favorite_Color = "Blue"
Student = True

print("Age = ",Age ,type(Age))
print("GPA = ",GPA,type(GPA))
print("Favorite_Color = ",Favorite_Color,type(Favorite_Color))
print("Student = ",Student,type(Student))

Age =  21 <class 'int'>
GPA =  4.0 <class 'float'>
Favorite_Color =  Blue <class 'str'>
Student =  True <class 'bool'>


3. Perform simple arithmetic:


In [19]:
num1 = 6
num2 = 7

sum = num1 + num2
print("Sum = ",sum)
print("Diff = ", num1 - num2)
print("Product = ", num1 * num2)
print("Division = ", num1 / num2)

Sum =  13
Diff =  -1
Product =  42
Division =  0.8571428571428571


4. String concatenation and f-strings:


In [6]:
First_name = "Thanitha"
Last_name = "Apiratmontree"
Full_name = First_name + " " + Last_name
print(f"Hello,{Full_name}")

Hello,Thanitha Apiratmontree


5. User input and type conversion:


In [20]:
num = input("Enter your favorite number : ")
print("type of input is : ", type(num))
num = int(num)
result = num + 5
print(num,"+ 5 = ", result)

Enter your favorite number : 4
type of input is :  <class 'str'>
4 + 5 =  9


Activity 1.1: Class Discussion - Uses of Scripting
Format: Open discussion, possibly in small groups first, then full class.
Prompt: "Based on the introduction, where do you think scripting with Python could be applied in real-world scenarios? Think about tasks you do daily or things you've seen in other courses."
Expected outcomes: Students will brainstorm and share ideas like automating file operations, simple data analysis, web development, game scripting, scientific computing, etc. Instructor will guide the discussion to highlight Python's versatility.


Answer : เขียน logic ของเกม , วิเคราะห์ข้อมูลจาก Excel เช่น คะแนนนักเรียน

Activity 1.2: "Python Puzzlers" - Identifying Data Types & Output
Format: Individual or pair work, followed by a class review.
Instructions: Provide a set of small Python code snippets (on a handout or projected). For each snippet, students must:
Predict the output.
Identify the data type of specific variables after certain operations.
Explain why they chose that output/data type.


In [9]:
# Puzzler 1
x = 10
y = 3.5
z = x + y
print(z)
print(type(z))


13.5
<class 'float'>


**ได้ผลลัพธ์:** 13.5
<class 'float'>

**ทำไมถึงเป็นtype float :** เพราะ x = 10 เป็น int และ y = 3.5 เป็น float เมื่อบวกกันจะได้ผลลัพธ์เป็น float (int + float = float)

In [10]:
# Puzzler 2
name = "Alice"
age = "30" # Note: This is a string
message = name + " is " + age + " years old."
print(message)
print(type(age))

Alice is 30 years old.
<class 'str'>


**ได้ผลลัพธ์:** Alice is 30 years old.
<class 'str'>

**ทำไมถึงเป็นtype :** เพราะ age = "30" ใส่เครื่องหมาย " " ทำให้ age เป็น string ไม่ใช่ int และการใช้ + ระหว่าง string คือการต่อข้อความ

In [11]:
# Puzzler 3
is_sunny = True
is_raining = False
print(is_sunny and is_raining)
print(type(is_sunny))

False
<class 'bool'>


**ได้ผลลัพธ์:** False
<class 'bool'>

**ทำไมถึงเป็นtype :** เพราะใช้ and ซึ่งจะได้Trueก็ต่อเมื่อ ทั้งสองค่าเป็นTrue และ type(is_sunny) คือ bool เพราะ True/False เป็น Boolean data type

In [16]:
# Puzzler 4
# What happens here?
 num_str = input("Enter a number: ")
 result = num_str + 5
 print(result)

IndentationError: unexpected indent (1707674877.py, line 3)

**ได้ผลลัพธ์:** Error

**ทำไมถึงเป็น :** เพราะ input() คืนค่าเป็น str เมื่อนำ str + int จะไม่สามารถรวมค่าได้เพราะ type ต่างกัน

In [13]:
num_str = input("Enter a number: ")
result = int(num_str) + 5
print(result)

Enter a number: 7
12
